lib/codecs/src/encoding/format/cef.rs

```diff
@@ -0,0 +1,541 @@
+use crate::encoding::BuildError;
+use bytes::BytesMut;
+use chrono::SecondsFormat;
+use lookup::lookup_v2::ConfigTargetPath;
+use snafu::Snafu;
+use std::num::ParseIntError;
+use std::{collections::HashMap, fmt::Write};
+use tokio_util::codec::Encoder;
+use vector_core::{
+    config::DataType,
+    event::{Event, LogEvent, Value},
+    schema,
+};
+
+const DEFAULT_DEVICE_VENDOR: &str = "Datadog";
+const DEFAULT_DEVICE_PRODUCT: &str = "Vector";
+const DEFAULT_DEVICE_VERSION: &str = "0"; // Major version.
+// TODO: find a way to get the actual vector version.
```

Comment:
Hmm, I can't recall if this information is hardcoded anywhere in the code. However, doing something like this, is a potential solution here: https://github.com/vectordotdev/vector/blob/master/lib/vector-vrl/web-playground/build.rs#L38-L42. I don't think we should block this PR but this would be a good followup. 

I think if we moved these comments [here](https://github.com/vectordotdev/vector/pull/17389/files#diff-ae2c2d1cbf6de991a10c399a96437795b6e3b065f1e9cc2638e6587e4f6a6e7aR211).

Reply:
It is possible to get the Vector version, though it may need to be injected into the codec. You can see how we fetch it to create the user-agent for the HTTP client here: https://github.com/vectordotdev/vector/blob/68c2a19b99fd4b1f297a022c781ecc5a8db3ff01/src/http.rs#L102-L103

Reply:
I can move both functions to the lib package and inject to codecs, just tried to keep all the refactoring out of the PR.

Reply:
Happy to see this in a followup PR 👍